docs/multipart_reference.rst

```diff
@@ -161,8 +161,12 @@ Multipart reference
 
    Multipart body writer.
 
+   ``boundary`` may be an ASCII-only string.
+
    .. attribute:: boundary
 
+      The byte-string representation of the boundary.
```

Comment:
The type bothers me.
We pass optional `str` value into constructor but the property is byteish.
Maybe the method should return `self._boundary.decode('ascii')`?
It is a breaking change but we could do it in aiohttp 3.0.
Amount of affected user code is very low IMHO.

Reply:
This behavior is unchanged. I just documented it.

https://github.com/aio-libs/aiohttp/pull/2545/files#diff-235eeadd1960735e1191879aa500b7b7L637

Reply:
Yes, I see.
But maybe now is the time for behavior changing?
Current one is confusing a little.

Reply:
Well, imo everything should be saved as byte-strings anyway (as I explained before), so having the boundary as a byte-string doesn't seem far off. It's only used that way, too, so either we'd add an additional property that converts the boundary from a unicode to a byte string, or we encode it at every point we need it (not a fan).

I'd say we leave this for now. I would rather spend time to convert all header strings to byte-strings rather than unwrapping the boundary here.

Reply:
I fear it was bytes all the time. Let's change that for 3.0! This could be done with separate PR with breaking change notice.

Reply:
Ok, let's merge.
Technically single Pull Request can have several records in `CHANGES/` but I'm ok with separate PR.

Reply:
@FichteFoll I have a feeling that all headers should be either strings or bytes.
All means really **all**, including `request.headers` etc.
This change in too intrusive, the ship has sailed many years ago.
Also I think 99% aiohttp users are happy with string headers but byte header values will confuse them.

Reply:
I agree that this should ideally be standardized on a single header type. For compatibility, all unicode strings *could* be converted to byte-strings using ASCII encoding, so that the outgoing API doesn't change much except that it allows byte-strings for all headers too.

However, this goes way beyond the scope of this PR and would be a large architectural consideration that I'm sure you're going to have a much better time deciding on. It might be something worth considering for the next major release, i.e. 3.0, as long as someone puts in the work.